test/tests/functional/pbs_ralter.py

```diff
@@ -1184,67 +1195,67 @@ def test_alter_degraded_resv_mom_down(self):
                                  alter_e=True, whichMessage=3, interactive=2,
                                  sequence=2)
         self.mom.start()
-
-    def test_sched_cycle_kicks_on_resv_end(self):
-        """
-        This test checks whether the sched cycle gets kicked
-        when the advance reservation ends
-        """
-        duration = 10
-        offset = 5
-
-        rid, _, _ = self.submit_and_confirm_reservation(
-            offset, duration, select="1:ncpus=8")
-        attr = {'Resource_List.walltime': '00:00:20'}
-        j = Job(TEST_USER)
-        j.set_sleep_time(10)
-        j.set_attributes(attr)
-        jid = self.server.submit(j)
-        self.server.expect(JOB, {'job_state': 'Q'},
-                           id=jid, max_attempts=5, interval=0.5)
-        self.scheduler.log_match(
-            jid + ";Job would conflict with reservation or top job",
-            n='ALL',  max_attempts=30, interval=0.5)
-        self.server.log_match(
-            rid.split('.')[0] + ";deleted at request of pbs_server",
-            id=rid.split('.')[0], n='ALL', max_attempts=30, interval=1)
-        self.server.log_match(jid + ";Job Run", id=jid,
-                              n='ALL', max_attempts=30, interval=0.5)
+        # test same for node offline case
+        attrs1 = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, attrs1, id=rid1)
+        self.server.expect(RESV, attrs1, id=rid2)
+        resv_node1 = self.server.status(RESV, {'resv_nodes'}, id=rid1)[0][
+            'resv_nodes'].split(':')[0].split('(')[1]
+        resv_node2 = self.server.status(RESV, {'resv_nodes'}, id=rid2)[0][
+            'resv_nodes'].split(':')[0].split('(')[1]
+        if resv_node1 is resv_node2:
+            self.server.manager(MGR_CMD_SET, NODE, {
+                                'state': "offline"}, id=resv_node1)
+        else:
+            self.server.manager(MGR_CMD_SET, NODE, {
+                                'state': "offline"}, id=resv_node1)
+            self.server.manager(MGR_CMD_SET, NODE, {
+                                'state': "offline"}, id=resv_node2)
+        self.server.expect(RESV, attrs, id=rid1)
+        self.server.expect(RESV, attrs, id=rid2)
+        self.alter_a_reservation(rid1, start1, end1, shift, alter_s=True,
+                                 alter_e=True, whichMessage=3, sequence=3)
+        self.alter_a_reservation(rid2, start2, end2, shift, alter_s=True,
+                                 alter_e=True, whichMessage=3, sequence=3)
```

Comment:
I don't think you'll get a denied message (whichMessage=3) if you don't have interactive set.  Same with the previous alterations to reservations.

Reply:
Hi @bhroam ,
There is no problem with the code. We get the denied message if interactive is not set.
For providing proof I printed the matched lines in the "alter_a_reservation" method:

`while attempts <= max_attempts:
                lines = self.server.log_match(msg, n='ALL', allmatch=True,
                                              interval=1, max_attempts=30)
                print lines`

and executed the degraded mom test. Following is the snippet of the logs:

_2017-10-06 10:12:13,326 INFO     expect on server centos1: reserve_state ~ RESV_DEGRADED|10 reservation R635.centos1 ...  OK
2017-10-06 10:12:13,326 INFOCLI  centos1: /opt/pbs/bin/pbs_rstat -f S636.centos1
2017-10-06 10:12:13,340 INFO     expect on server centos1: reserve_state ~ RESV_DEGRADED|10 reservation S636.centos1 ...  OK
2017-10-06 10:12:13,341 INFO     reservation alter on centos1: R635.centos1 {'reserve_start': '201710061013.12', 'reserve_end': '201710061013.42'}
2017-10-06 10:12:13,341 INFOCLI  centos1: /opt/pbs/bin/pbs_ralter -R 201710061013.12 -E 201710061013.42 R635.centos1
2017-10-06 10:12:13,368 INFO     pbs_ralter: R635.centos1 ALTER REQUESTED displayed
2017-10-06 10:12:13,370 INFO     server centos1 log match: searching for "Resv;R635.centos1;Attempting to modify reservation start=Fri Oct 06 10:13:12 2017 end=Fri Oct 06 10:13:42 2017"... OK
2017-10-06 10:12:13,370 INFO     server centos1 log match: searching for "Resv;R635.centos1;Reservation alter denied" - on all matches ... OK
[(404, '10/06/2017 10:12:10;0200;Server@centos1;Resv;R635.centos1;Reservation alter denied\n'), (436, '10/06/2017 10:12:12;0200;Server@centos1;Resv;R635.centos1;Reservation alter denied\n')]
2017-10-06 10:12:13,370 INFO     log_match: searching 3 sequence of message: Resv;R635.centos1;Reservation alter denied: Got: 2
2017-10-06 10:12:13,873 INFO     server centos1 log match: searching for "Resv;R635.centos1;Reservation alter denied" - on all matches ... OK
**[(404, '10/06/2017 10:12:10;0200;Server@centos1;Resv;R635.centos1;Reservation alter denied\n'), (436, '10/06/2017 10:12:12;0200;Server@centos1;Resv;R635.centos1;Reservation alter denied\n'), (501, '10/06/2017 10:12:13;0200;Server@centos1;Resv;R635.centos1;Reservation alter denied\n')]**_

If you check the last bold italics line, it has the denied message in the list of matched lines printed.

Reply:
OK, I see the code I was missing.  There's one part of alter_a_reservation() that checks for the interactive return of the command and then another second that checks for the log message.  You are correct in your alter_a_reservation() usage.